microprofile.jdt/org.eclipse.lsp4mp.jdt.core/src/main/java/org/eclipse/lsp4mp/jdt/internal/faulttolerance/java/MicroProfileFaultToleranceDiagnosticsParticipant.java

```diff
@@ -103,11 +103,37 @@ public boolean visit(MethodDeclaration node) {
 			try {
 				validateMethod(node, diagnostics, context);
 			} catch (JavaModelException e) {
-				LOGGER.log(Level.WARNING, "An exception occurred when attempting to validate the fallback method");
+				LOGGER.log(Level.WARNING, "An exception occurred when attempting to validate the annotation marked method");
 			}
 			super.visit(node);
 			return true;
 		}
+		
+		@Override
+		public boolean visit(TypeDeclaration type) {
+			// Retrieve annotation(s) marked on class
+			@SuppressWarnings("rawtypes")
+			List modifiers = type.modifiers();
+			for (Object modifier : modifiers) {
+				if (modifier instanceof Annotation) {
+					Annotation annotation = (Annotation) modifier;
+					if (isMatchAnnotation(annotation, ASYNCHRONOUS_ANNOTATION)) {
+						try {
+							MethodDeclaration[] methods = type.getMethods();
+							for (MethodDeclaration node : methods) {
+								MarkerAnnotation markAnnotation = (MarkerAnnotation) modifier;
+								validateAsynchronousAnnotation(node, diagnostics, context, markAnnotation);
```

Comment:
I would delete `markerAnnotation` and just pass `(MarketAnnotation) modifier` in as an argument